### PR TITLE
Include tokens in flex flow run info

### PR DESCRIPTION
**Pull request by lumoslnt**
# Description

Include tokens in flex flow run info.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-core/promptflow/_core/run_tracker.py [hunk 1]
```diff
@@ -392,13 +392,12 @@ def set_inputs(self, run_id: str, inputs: Mapping[str, Any]):
         run_info = self.ensure_run_info(run_id)
         run_info.inputs = inputs
 
-    def set_openai_metrics(self, run_id: str):
+    def set_openai_metrics(self, run_id: str, api_calls: List):
```
Comment (thy09) [inline src/promptflow-core/promptflow/_core/run_tracker.py:395]: We can simply do not use run_id for this scenario?

File: src/promptflow-core/promptflow/executor/_script_executor.py [hunk 2]
```diff
@@ -184,6 +185,7 @@ async def _exec_line_async(
             output = self._stringify_generator_output(output) if not allow_generator_output else output
             traces = Tracer.end_tracing(line_run_id)
             output_dict = convert_eager_flow_output_to_dict(output)
+            run_tracker.set_openai_metrics(line_run_id, traces)
```
Comment (thy09) [inline src/promptflow-core/promptflow/executor/_script_executor.py:188]: Remember to add test cases.

File: src/promptflow-core/promptflow/_core/run_tracker.py (hunk 1 above)
Comment (lumoslnt) [reply src/promptflow-core/promptflow/_core/run_tracker.py]: Refined the code to avoid changing the interface.

**Review by Adrian999999999 — APPROVED** [review]

**Review by thy09 — DISMISSED** [review]

File: src/promptflow-core/promptflow/executor/_script_executor.py (hunk 2 above)
Comment (lumoslnt) [reply src/promptflow-core/promptflow/executor/_script_executor.py]: Added.

**Review by thy09 — APPROVED** [review]

**Review by D-W- — APPROVED** [review]

File: src/promptflow-core/tests/core/e2etests/test_eager_flow.py [hunk 3]
```diff
@@ -87,6 +90,24 @@ def test_flow_run(self, flow_folder, inputs, ensure_output, init_kwargs):
...
         assert line_result1.output == line_result2.output
 
+    def test_flow_run_with_openai_chat(self, dev_connections):
+        flow_file = get_yaml_file("callable_class_with_openai", root=EAGER_FLOW_ROOT, file_name="flow.flex.yaml")
+        connection = dev_connections["azure_open_ai_connection"]
+        # TODO: Remove this after the connection type is added to github secrets
+        if "type" not in connection:
+            connection["type"] = "AzureOpenAIConnection"
+
+        with patch(
```
[the remark this answers is not in the document]
Comment (lumoslnt) [reply src/promptflow-core/tests/core/e2etests/test_eager_flow.py:100]: Will do.

**Pull request merged**